pkgs/development/libraries/mozc/default.nix

```
@@ -0,0 +1,234 @@
+      hash = "sha256-7CfIr7hJChBWFVynCVqw5KYh/lIjl277fqQQ4AuJ7x4=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-alt-cannadic";
+      name = repo;
+      rev = "f59287e569db3e226378380a34e71275654b46d0";
+      hash = "sha256-a9U6mGlGAxbywILeAaWKbt7BFWRPFS+UZvUhliFUseY=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-edict2";
+      name = repo;
+      rev = "e0cbf7d3192b1cdd38629a720b295bcbd67cd8bd";
+      hash = "sha256-jj5eBcmWxLxKC/Q720BPuzxJc9x1Y2RvY0kJfBJkIfE=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-jawiki";
+      name = repo;
+      rev = "be4da36c087e56a3aa5f835f8d9e810fc1a060ba";
+      hash = "sha256-WkP/RugY28TqNwTjbi2V5BsRVLwUIIvOX6zGJmdIDEk=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-neologd";
+      name = repo;
+      rev = "90e59c7707a5fe250c992c10c6ceb08a7ce7e652";
+      hash = "sha256-zY7K/J4OzBTQHrj8sF4s8xPqakoWHHMxWrvnvHT6oxE=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-personal-names";
+      name = repo;
+      rev = "0e51584283f759cb6bfac1f19eda92c2c524eed4";
+      hash = "sha256-8cJmSp7rGxSHy+TElFpkXmNvMxYLw2yiAYJJb2AUyyE=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-place-names";
+      name = repo;
+      rev = "478004504981e094af407f4fbd64cd65cf2d85dd";
+      hash = "sha256-3BwK+RuZMl5+femxQXNhDF/NE3qfX6u/+imqL2Uqd8I=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-skk-jisyo";
+      name = repo;
+      rev = "43518e6ea033681580a515281668c85eb74a5b14";
+      hash = "sha256-05T//ulsS5HvOKPdOEG87/Yp8GgzOB2X3wG8Sds3uUU=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-sudachidict";
+      name = repo;
+      rev = "1b5ad7cb51325fc9b42732887fede29d6a02bb07";
+      hash = "sha256-E4kq3YnxdYRFlv4KLSvf4NRoPhWXL8c89JIvqsjuy3o=";
+    })
+    (fetchurl rec {
+      name = "jawiki";
+      url = "https://dumps.wikimedia.org/${name}/20230820/${name}-20230820-all-titles-in-ns0.gz";
+      recursiveHash = true;
+      hash = "sha256-pOK4ThhOaup4t896Wx9jfsKSmWlQYn4Yy4pTiiwAFIs=";
+      downloadToTemp = true;
+      postFetch = ''
+        mkdir -p $out
+        mv $downloadedFile $out/${name}.gz
+      '';
+    })
+    (fetchFromGitHub rec {
+      owner = "musjj";
+      repo = "jp-zip-codes";
+      name = repo;
+      rev = "be9cd2ac888a34eba30a425c43a7e54f5187a649";
+      hash = "sha256-02CE/DJnqNXUF0OH3ptq2sb8UbtqKjLKx7Hf0izq+X4=";
+    })
+  ];
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    glibcLocales
+    pkg-config
+    python3
+    ruby
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [ qtbase ];
+
+  preBuild = ''
+    cd mozc/src
+  '';
+
+  LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
+  LC_ALL = "en_US.UTF-8";
```

```suggestion
  env.LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
  env.LC_ALL = "en_US.UTF-8";
```
Shall also add a comment on why they are set.